crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -615,14 +630,18 @@ impl<
                             .insert(package_name.clone(), UnavailablePackage::Offline);
                         return Ok(None);
                     }
-                    MetadataResponse::InvalidMetadata(_) => {
-                        self.unavailable_packages
-                            .insert(package_name.clone(), UnavailablePackage::InvalidMetadata);
+                    MetadataResponse::InvalidMetadata(err) => {
+                        self.unavailable_packages.insert(
+                            package_name.clone(),
+                            UnavailablePackage::InvalidMetadata(err.to_string()),
+                        );
                         return Ok(None);
                     }
-                    MetadataResponse::InvalidStructure(_) => {
-                        self.unavailable_packages
-                            .insert(package_name.clone(), UnavailablePackage::InvalidStructure);
+                    MetadataResponse::InvalidStructure(err) => {
+                        self.unavailable_packages.insert(
+                            package_name.clone(),
+                            UnavailablePackage::InvalidStructure(err.to_string()),
```

Comment:
It's confusing but for URLs, we don't have a version... so both `UnavailablePackage` and `IncompletePackage` have an `InvalidStructure` and `InvalidMetadata` variant.